docs/ir/ir-breaking-changes.rst

```diff
@@ -91,29 +91,71 @@ This causes differences in some contracts, for example:
 This causes differences in some contracts, for example:
 ::
-     // SPDX-License-Identifier: GPL-3.0
-     pragma solidity >0.8.0;
-
-     contract C {
-         bytes x;
-         function f() public returns (uint r) {
-             bytes memory m = "tmp";
-             assembly {
-                 mstore(m, 8)
-                 mstore(add(m, 32), "deadbeef15dead")
-             }
-             x = m;
-             assembly {
-                 r := sload(x.slot)
-             }
-         }
-     }
-
-Previously `f()` would return `0x6465616462656566313564656164000000000000000000000000000000000010` (it has correct length, and correct first 8 elements, but than it contains dirty data which was set via assembly).
-Now it is returning `0x6465616462656566000000000000000000000000000000000000000000000010` (it has correct length, and correct elements, but doesn't contain dirty data).
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity >0.8.0;
+
+    contract C {
+        bytes x;
+        function f() public returns (uint _r) {
+            bytes memory m = "tmp";
+            assembly {
+                mstore(m, 8)
+                mstore(add(m, 32), "deadbeef15dead")
+            }
+            x = m;
+            assembly {
+                _r := sload(x.slot)
+            }
+        }
+    }
+
+Previously ``f()`` would return ``0x6465616462656566313564656164000000000000000000000000000000000010`` (it has correct length, and correct first 8 elements, but then it contains dirty data which was set via assembly).
+Now it is returning ``0x6465616462656566000000000000000000000000000000000000000000000010`` (it has correct length, and correct elements, but does not contain superfluous data).
+
+.. index:: ! evaluation order; expression
+
+* For the old code generator, the evaluation order of expressions is unspecified.
+  For the new code generator, we try to evaluate in source order (left to right), but do not guarantee it.
+  This can lead to semantic differences.
+
+For example:
+
+::
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity >0.8.0;
+    contract C {
+        function preincr_u8(uint8 _a) public pure returns (uint8) {
+            return ++_a + _a;
+        }
+    }
+
+The function ``preincr_u8(1)`` returns the following values:
+- Old code generator: 3 (``1 + 2``) but the return value is unspecified in general
+- New code generator: 4 (``2 + 2``) but the return value is not guaranteed
+
+.. index:: ! evaluation order; function arguments
+
+On the other hand, function argument expressions are evaluated similarly by old as well as new code generators.
+For example:
+
+::
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity >0.8.0;
+    contract C {
+        function add(uint8 _a, uint _b) public pure returns (uint8) {
```

Comment:
```suggestion
        function add(uint8 _a, uint8 _b) public pure returns (uint8) {
```

I think this is the cause of the CI failure: https://app.circleci.com/pipelines/github/ethereum/solidity/15181/workflows/6fcbaf6b-0d56-4cf9-92a9-9da8be9d7aee/jobs/690885